pkg/extensions/brokers/natsjetstream/natsjetstream.go

```
@@ -0,0 +1,245 @@
+	return func(controller *ControllerOption) error {
+		if config.Name == "" {
+			return fmt.Errorf("stream name is required")
+		}
+		controller.streamName = config.Name
+
+		if _, err := controller.jetStream.CreateStream(context.Background(), config); err != nil {
+			if !errors.Is(err, jetstream.ErrStreamNameAlreadyInUse) {
+				return fmt.Errorf("could not create stream: %w", err)
+			}
+			if _, err = controller.jetStream.UpdateStream(context.Background(), config); err != nil {
+				return fmt.Errorf("could not create or update stream: %w", err)
+			}
+		}
+
+		return nil
+	}
+}
+
+// WithStream uses the given stream name (the stream has to be created before initializing)
+func WithStream(name string) ControllerOptionOption {
+	return func(controller *ControllerOption) error {
+		controller.streamName = name
+		return nil
+	}
+}
+
+// WithConsumerConfig creates or updates a consumer based on the given consumer configuration.
+func WithConsumerConfig(config jetstream.ConsumerConfig) ControllerOptionOption {
+	return func(controller *ControllerOption) error {
+		if config.Name == "" {
+			return fmt.Errorf("consumer name is required")
+		}
+		controller.consumerName = config.Name
+
+		if _, err := controller.jetStream.CreateOrUpdateConsumer(context.Background(), controller.streamName, config); err != nil {
+			return fmt.Errorf("could not create or update consumer: %w", err)
+		}
+
+		return nil
+	}
+}
+
+// WithConsumer uses the given consumer name (the consumer has to be created before initializing).
+func WithConsumer(name string) ControllerOptionOption {
+	return func(controller *ControllerOption) error {
+		controller.consumerName = name
+		return nil
+	}
+}
+
+// NewController creates a new NATS JetStream controller.
+func NewController(url string, options ...ControllerOptionOption) *ControllerOption {
+	// Connect to NATS
+	nc, err := nats.Connect(url)
+	if err != nil {
+		panic(err)
```

As we don't know how people could use it, I think it's better to return an error on this function instead of the panics.

---

Returning an error here led to a breaking change in the API, but I assume this is okayish.
I've adjusted all controllers and the documentation as well.
https://github.com/lerenn/asyncapi-codegen/pull/121/commits/2b47ac6d488538b7edbcce61da3ec72f8851e34b

---

This is still bit "beta" code, so those kind of change is still acceptable (not a full rewrite of course).